impeller/renderer/allocator.cc

```diff
@@ -45,6 +45,14 @@ std::shared_ptr<DeviceBuffer> Allocator::CreateBuffer(
   return OnCreateBuffer(desc);
 }
 
+uint64_t Allocator::GetAllocatedSize() const {
+  return allocated_size_;
+}
+
+void Allocator::ResetAllocatedSize() {
+  allocated_size_ = 0;
+}
+
```

Comment:
Perhaps we should add/remove these with some ifdef.

Reply:
Another idea is that if we do this only in builds for engine unit tests, we could probably track a vector of allocated sizes which make assertions easier